diff -r 0285b706fcde lrm/lrmd/lrmd.c
--- a/lrm/lrmd/lrmd.c	Tue Sep 28 19:10:38 2010 +0200
+++ b/lrm/lrmd/lrmd.c	Thu Feb 02 20:27:33 2012 +1100
@@ -832,6 +832,13 @@ main(int argc, char ** argv)
 		init_stop(PID_FILE);
 	}
 
+        if(getenv("LRMD_MAX_CHILDREN")) {
+            int tmp = atoi(getenv("LRMD_MAX_CHILDREN"));
+            if(tmp > 4) {
+                max_child_count = tmp;
+            }
+        }
+        
 	return init_start();
 }
 

