diff --git include/glue_config.h.in include/glue_config.h.in
index a139da0..017aa19 100644
--- include/glue_config.h.in
+++ include/glue_config.h.in
@@ -1,8 +1,5 @@
 /* include/config.h.in.  Generated from configure.in by autoheader.  */
 
-/* Location for daemons */
-#undef GLUE_DAEMON_DIR
-
 /* Group to run daemons as */
 #undef GLUE_DAEMON_GROUP
 
@@ -21,9 +18,6 @@
 /* Group to run daemons as */
 #undef HA_APIGROUP
 
-/* Location for daemons */
-#undef HA_LIBHBDIR
-
 /* top directory of area to drop core files in */
 #undef HA_COREDIR
 
@@ -33,12 +27,6 @@
 /* Default logging facility */
 #undef HA_LOG_FACILITY
 
-/* Default plugin search path */
-#undef PILS_BASE_PLUGINDIR
-
-/* Where to find plugins */
-#undef HA_PLUGIN_DIR
-
 /* Location of system configuration files */
 #undef HA_SYSCONFDIR
 
@@ -59,9 +47,6 @@
 /* Location for v1 Heartbeat RAs */
 #undef HB_RA_DIR
 
-/* Where to find LRM plugins */
-#undef LRM_PLUGIN_DIR
-
 /* Location for LSB RAs */
 #undef LSB_RA_DIR
 
@@ -80,15 +65,6 @@
 /* Current pacemaker version */
 #undef GLUE_VERSION
 
-/* Location of non-pluing stonith scripts */
-#undef STONITH_EXT_PLUGINDIR
-
-/* Location of RHCS stonith scripts */
-#undef STONITH_RHCS_PLUGINDIR
-
-/* Location of stonith plugins */
-#undef STONITH_MODULES
-
 /* Stonith plugin domain */
 #undef ST_TEXTDOMAIN
 
diff --git lrm/test/LRMBasicSanityCheck.in lrm/test/LRMBasicSanityCheck.in
index dbe8548..dd8cc55 100755
--- lrm/test/LRMBasicSanityCheck.in
+++ lrm/test/LRMBasicSanityCheck.in
@@ -17,8 +17,6 @@
  # License along with this library; if not, write to the Free Software
  # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  #
-HBLIB=@libdir@/heartbeat
-LRMD=$HBLIB/lrmd
 LRMADMIN=@sbindir@/lrmadmin
 export LRMD LRMADMIN
 
